### PR TITLE
Adds GetSignInChallenge flow to sign in flow

### PR DESCRIPTION
**Pull request by arshmalhotra**
First half of sign in flow to get challenge (user's password salt, nonce)

### CODE REVIEW

File: Backend/routes/auth.py [hunk 1]
```diff
@@ -1,10 +1,34 @@
...
+from flask import Blueprint, render_template, session, abort, request, make_response
+from services.authentication import sign_up_service
+from services.authentication.sign_in_service import SignInService
+
+get_sign_in_challenge = Blueprint('get_sign_in_challenge', __name__)
+@get_sign_in_challenge.route('/get_sign_in_challenge', methods=['POST'])
+def get_sign_in_challenge_route():
+    ############
+    # switch to request.form for POST
+    ############
```
Comment (arshmalhotra) [inline Backend/routes/auth.py]: Get rid of comment

File: Backend/routes/auth.py [hunk 2]
```diff
@@ -1,10 +1,34 @@
...
+        .executeGetSignInChallengeRequest(challengeRequest)
+    res = challengeResponse.toFlaskResponse()
+
+    return res
 
 sign_in = Blueprint('sign_in', __name__)
 @sign_in.route('/sign_in')
 def sign_in_route():
-    return sign_in_service.sign_in()
+
```
Comment (arshmalhotra) [inline Backend/routes/auth.py]: Add return or pass

**Review by pramesh99 — DISMISSED** [review]
LGTM

**Review by pramesh99 — APPROVED** [review]
LGTM

**Pull request merged**